File: Day02/Part02/day02_02.py
given_file = open("input.txt", "r")
ans = 0

# get list of integers from each line
nums = list(map(int, given_file.readline().split()))

while nums:

    # get result of all numbers that evenly divide
    mini_nums = [a // b for a in nums for b in nums if a != b and a % b == 0]

    # should only be one per line
    ans += mini_nums[0]

    # get next line
    nums = list(map(int, given_file.readline().split()))

# print answer
print("Answer:", ans)

# close file
given_file.close()

File: Day02/Part02/day02_02.java
import java.io.*;
import java.util.*; // Required for Arrays.stream()


public class day02_02 {

    public static void main(String[] args) {

        Integer ans = 0;

        try {

            BufferedReader br = new BufferedReader(new FileReader("input.txt"));

            String line = br.readLine();

            while (line != null) {

                // map array of strings to integers
                Integer[] nums = Arrays.stream(line.split("\\s+")).map(Integer::valueOf).toArray(Integer[]::new);

                for (Integer i : nums) {
                    for (Integer j : nums) {
                        if (i != j && i % j == 0) {
                            // assuming only one answer per line
                            ans += (i / j);
                        }
                    }
                }
                
                // get next line
                line = br.readLine();
            }

            br.close();

        } catch(IOException e) {
            e.printStackTrace();
        }

        System.out.println("Answer: " + String.valueOf(ans));

    }

}
